neurodamus/node.py

```diff
@@ -375,14 +375,14 @@ def compute_load_balance(self):
             return None
         elif lb_mode == LoadBalanceMode.Memory:
             logging.info("Load Balancing ENABLED. Mode: Memory")
-            alloc = import_allocation_stats("allocation.pkl.gz")
+            alloc = import_allocation_stats("allocation.pkl.gz", self._cycle_i)
             for pop, ranks in alloc.items():
                 for rank, gids in ranks.items():
                     logging.debug(f"Population: {pop}, Rank: {rank}, Number of GIDs: {len(gids)}")
             if MPI.rank == 0:
                 unique_ranks = set(rank for pop in alloc.values() for rank in pop.keys())
                 logging.debug("Unique ranks in allocation file: %s", len(unique_ranks))
-                if MPI.size != len(unique_ranks):
+                if MPI.size < len(unique_ranks):
```

Comment:
Do we want to proceed knowing the resources will be under-utilized?
Either way, since we store the information about METypes we can redistribute cells very quickly. So are we leaving that for the next iteration and for the moment we simply error out?

Reply:
I resorted to using the `<` since the error can be a bit annoying on the user side if the number if ranks is not exactly the same. Would be cool to be able to redistribute on the fly if the number of ranks is not exactly the same though.

I'll revert to the old `!=` for the moment, in any case, then we can add this at a later stage.